x509/verify.go

```diff
@@ -198,7 +209,7 @@ func (c *Certificate) buildChains(cache map[int][][]*Certificate, currentChain [
 		if err != nil {
 			continue
 		}
-		if !root.inChain(currentChain) {
+		if !root.subjectAndSPKIInChain(currentChain) {
```

Comment:
I'm not sure. I didn't merge this based on this comment, but I'm not sure if this is blocking or not. Can you resolve one way or another?